### ISSUE TITLE
Citation Key is overwritten when importing bibtex code with ctrl+v even when the generator is disabled.

### ISSUE THREAD

**Issue by MKHoffmann** [JabRef/jabref#9455]
### JabRef version

5.7 (latest release)

### Operating system

Windows

### Details on version and operating system

JabRef 5.7--2022-08-05--73c111c Windows 11 10.0 amd64  Java 18.0.2  JavaFX 18.0.1+2

### Checked with the latest development build

- [X] I made a backup of my libraries before testing the latest development version.
- [X] I have tested the latest development version and the problem persists

### Steps to reproduce the behaviour

1. Disable Citation Key Generator
2. Copy a bibtex citation code from Google Scholar
3. Import it using ctrl+v
4. Citation Key is changed to something different.

**Comment by ThiloteE** [comment]
With "disable Citation Key Generator" you mean to disable "`Overwrite existing keys`" in the preferences?
![image](https://user-images.githubusercontent.com/73715071/208052912-bd01bbf3-8dbd-4aa4-8f77-5d447914fd21.png)

**Comment by ThiloteE** [comment]
Related issues/PRs:

- https://github.com/JabRef/jabref/pull/8361 (Last major rework of citationkey behaviour)
- https://github.com/JabRef/jabref/issues/8406

**Comment by koppor** [comment]
We think, our preferences are not well organized in this respect. Please check following setting

![grafik](https://user-images.githubusercontent.com/1366654/208191688-22d78d23-6d61-473d-a7b0-c6b84d40860c.png)

**Comment by koppor** [comment]
Since requires internal refactorings, we postpone this after the release of v5.8.

**Comment by MKHoffmann** [comment]
Yes, all these settings are disabled. I should add, I am using Jabref on a Windows 11 machine.

**Comment by koppor** [comment]
Our test case:

Library without following entry:

```bibtex
@Article{test-key-not-generated,
  author = {Test Author},
}
```

1. Use JabRef build as of today (not 5.7)
2. Run on Windows 10
3. Copy the entry from GitHub to the clipbard
4. Paste entry

Result:

![grafik](https://user-images.githubusercontent.com/1366654/208299516-d38cce28-eff9-4c17-9c94-108c9df13888.png)

Video of it https://www.loom.com/share/d5098caf604e456296169fa68841d09e

**Comment by MKHoffmann** [comment]
For version 5.8 and Windows 11: If I paste your test case, this is my output:
![image](https://user-images.githubusercontent.com/50181107/208381666-8d37e6c0-312e-49b3-81af-928b252a219c.png)
With the settings:
![image](https://user-images.githubusercontent.com/50181107/208381964-dad3254a-692c-4363-a923-5e55aef11f97.png)

**Comment by koppor** [comment]
Oh, wow. Can you

1. Export your preferences  (small arrow on the lower left side on the dialog) - and send them to us (`web -> jabref.org`)
2. Reset the preferences (if that is OK for you)
3. Change both (!) preferences again. The thing is, in your case, JabRef DOES NOT check "Overwrite existing keys", but uses "Generate new key on for imported antries". In other words: It does not matter what you ticket at "Overwrite existing keys", because this is about the behavior of the manual triggering of the key generator, not the logic running when importing an entry.
4. Check again

**Comment by MKHoffmann** [comment]
Sure, I will do that, how should I send it though? Via the forum or what is the preferred way?

**Comment by MKHoffmann** [comment]
Thank you, I sent the preference settings and tried your steps, the problem still occurs.

**Comment by Siedlerchr** [comment]
Maybe related issue? https://github.com/JabRef/jabref/issues/8644

**Comment by MKHoffmann** [comment]
Yes, that helped, thank you!

**Comment by calixtus** [comment]
Current layout in the prefs:

<img width="611" height="318" alt="Image" src="https://github.com/user-attachments/assets/3efca473-b7e3-440c-bcd6-13a4322e5f52" />

Is this issue still valid?

**Comment by calixtus** [comment]
After discussion we believe that this issue was fixed by adressing issue #12436 .
The preference options should make it clear to the user, that keys will be overwritten if "Generate a new key for imported entries (overwriting their default)" will overwrite their custom citation keys.
